routes/api/v1/repo/repo.go

```diff
@@ -69,7 +70,7 @@ func Search(c *context.APIContext) {
 		results[i] = repos[i].APIFormat(nil)
 	}
 
-	c.SetLinkHeader(int(count), setting.API.MaxResponseItems)
+	c.SetLinkHeader(int(count), opts.PageSize)
```

Comment:
We should check and make sure `opts.PageSize` to not larger than `setting.API.MaxResponseItems`.